docs/resources/azurerm_virtual_machines.md.erb

```diff
@@ -0,0 +1,120 @@
+---
+title: About the azurerm_virtual_machines Resource
+platform: azure
+---
+
+# azurerm\_virtual\_machines
+
+Use the `azurerm\_virtual\_machines` InSpec audit resource to test properties
+related to virtual machines for a resource group.
+
+<br />
+
+## Syntax
+
+The `resource_group` must be given as a parameter.
+
+    describe azurerm_virtual_machines(resource_group: 'MyResourceGroup') do
+      ...
+    end
+
+## Examples
+
+    # Exists if any virtual machines exist in the resource group
+    describe azurerm_virtual_machines(resource_group: 'MyResourceGroup') do
+      it { should exist }
+    end
+
+    # Filters based on platform
+    describe azurerm_virtual_machines(resource_group: 'MyResourceGroup').where('platform' => 'windows') do
+      it { should exist }
+    end
+
+<br />
+
+## Parameters
+
+* `resource_group`
+
+### Parameter Examples
+
+### resource_group (required)
+
+    describe azurerm_virtual_machines(resource_group: 'MyResourceGroup') do
+      ...
+    end
+
+<br />
+
+## Filter Criteria
+
+* `platform`, `name`, `os_disk`
+
+### platform
+
+Filters the results to only include those that match the given platform. Valid
+choices are `linux` and `windows`.
+
+    # Insist that you have at least one windows virtual machine
+    describe azurerm_virtual_machines(resource_group: 'MyResourceGroup').where('platform' => 'windows') do
+      it { should exist }
+    end
+
+### name
+
+Filters the result to only those that match the given name.
+
+    # Insist that you have at least one virtual machine that starts with 'MyVm'
+    describe azurerm_virtual_machines(resource_group: 'MyResourceGroup').where { name.start_with?('WindowsVm') } do
+      it { should exist }
+    end
+
+## Properties
+
+* `os_disks`, `data_disks`, `vm_names`
+
+### os_disks
+
+Gives a list of OS disk names for all the virtual machines in the resource group.
+
+    its('os_disks.sort') { should eq ['MyDisk1', 'MyDisk2'] }
+
+### data_disks
+
+Gives a list of data disks for all the virtual machines in the resource group.
+
+    its('os_disks.sort') { should eq [['MyDisk1'], ['MyDisk2']] }
+
+Note that it returns an array of arrays. You may flatten them for testing:
+
+    its('os_disks.flatten.sort') { should eq ['MyDisk1', 'MyDisk2'] }
```

Comment:
Could we give examples that avoid `.sort`, like:

```
its('os_disks') { should include 'DataDisk1' }
its('os_disks') { should include 'DataDisk2' }
```

Reply:
This one is tricky as we have an array of arrays here. The underlying structure is:
```
[['MyDisk1', 'MyDisk2'], ['MyDisk3']]
```
Do you have a recommendation on how to do this more cleaning in the filter table implementation?

Reply:
Hmmm...not at the moment, you're right. Array of array's is tricky. Disregard my comment :smile:.